File: Orimath.ViewPlugins/Plugins/IViewPlugin.fs
namespace Orimath.Plugins

type ViewPluginArgs =
    { Workspace: IWorkspace
      FileManager: IFileManager
      Messenger: IMessenger
      Dispatcher: IDispatcher
      PointConverter: IViewPointConverter }


type IViewPlugin =
    abstract member Execute : args: ViewPluginArgs -> unit

File: Orimath.ViewPlugins/Plugins/IViewPlugin.cs
namespace Orimath.Plugins
{
    public class ViewPluginArgs
    {
        public IWorkspace Workspace { get; }
        public IMessenger Messenger { get; }
        public IDispatcher Dispatcher { get; }
        public IViewPointConverter PointConverter { get; }

        public ViewPluginArgs(IWorkspace workspace, IMessenger messenger, IDispatcher dispatcher, IViewPointConverter pointConverter)
        {
            Workspace = workspace;
            Messenger = messenger;
            Dispatcher = dispatcher;
            PointConverter = pointConverter;
        }
    }

    public interface IViewPlugin
    {
        void Execute(ViewPluginArgs args);
    }
}
